### PR TITLE
Feat/use group path for seating assets

### PR DESCRIPTION
**Pull request by dpanta94**
### 🎫 Ticket

[SL-246]

### 🗒️ Description

Affected assets (for QA):

Admin assets
```php
$assets = [
			'tec-tickets-seating-admin-maps'              => '/build/Seating/admin/maps.js',
			'tec-tickets-seating-admin-maps-style'        => '/build/Seating/admin/maps.css',
			'tec-tickets-seating-admin-layouts'           => '/build/Seating/admin/layouts.js',
			'tec-tickets-seating-admin-layouts-style'     => '/build/Seating/admin/layouts.css',
			'tec-tickets-seating-admin-map-edit'          => '/build/Seating/admin/mapEdit.js',
			'tec-tickets-seating-admin-map-edit-style'    => '/build/Seating/admin/mapEdit.css',
			'tec-tickets-seating-admin-layout-edit'       => '/build/Seating/admin/layoutEdit.js',
			'tec-tickets-seating-admin-layout-edit-style' => '/build/Seating/admin/layoutEdit.css',
		];
```

Utils assets
```php
$assets = [
			'tec-tickets-seating-service-bundle' => '/build/Seating/service.js',
			'tec-tickets-seating-utils'          => '/build/Seating/utils.js',
			'tec-tickets-seating-currency'       => '/build/Seating/currency.js',
		];
```

Block Editor Assets
```php
$assets = [
			'tec-tickets-seating-block-editor'       => '/build/Seating/blockEditor.js',
			'tec-tickets-seating-block-editor-style' => '/build/Seating/blockEditor.css',
		];
```
Frontend assets
```php
$assets = [
			'tec-tickets-seating-frontend'       => '/build/Seating/frontend/ticketsBlock.js',
			'tec-tickets-seating-frontend-style' => '/build/Seating/frontend/ticketsBlock.css',
		];
```
Ajax assets
```php
$assets = [
			'tec-tickets-seating-ajax' => '/build/Seating/ajax.js',
		];
```
Timer assets
```php
$assets = [
			'tec-tickets-seating-session'       => '/build/Seating/frontend/session.js',
			'tec-tickets-seating-session-style' => '/build/Seating/frontend/session.css',
		];
```
Order Assets
```php
$assets = [
			'tec-tickets-seating-admin-seats-report'       => '/build/Seating/admin/seatsReport.js',
			'tec-tickets-seating-admin-seats-report-style' => '/build/Seating/admin/seatsReport.css',
		];
```
### ✔️ Checklist
- [x] Ran `npm run changelog` to add changelog file(s). More info [here](https://docs.theeventscalendar.com/developer/git/changelogs/#process)
- [x] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [ ] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [ ] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [ ] Add your PR to the project board for the release.

[SL-246]: https://stellarwp.atlassian.net/browse/SL-246?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/slr_integration/Frontend/Timer_Test.php [hunk 1]
```diff
@@ -69,6 +69,31 @@ public function reset_cookie(): void {
...
 	}
 
+	public function asset_data_provider() {
+		$assets = [
+			'tec-tickets-seating-session'       => '/build/Seating/frontend/session.js',
+			'tec-tickets-seating-session-style' => '/build/Seating/frontend/session.css',
+		];
+
+		foreach ( $assets as $slug => $path ) {
+			yield $slug => [ $slug, $path ];
```
Comment (lucatume) [inline tests/slr_integration/Frontend/Timer_Test.php:79]: If you use a Generator and yield from an array, you might as well return an array.
PHPUnit will eagerly resolve the Generator anyway to an array anyway.

If you want to use the Generator, then just `yield` twice:

```php
public function asset_data_provider(): Generator {
  yield 'tec-tickets-seating-session' => '/build/Seating/Frontend/session.js`;
  yield 'tec-tickets-seating-session-style' => '/build/Seating/Frontend/session.css`;
}
```

File: src/Tickets/Seating/Orders/Controller.php [hunk 2]
```diff
@@ -531,9 +528,11 @@ public function get_localized_data( ?int $post_id = null ): array {
 	private function register_assets(): void {
 		Asset::add(
 			'tec-tickets-seating-admin-seats-report',
-			$this->built_asset_url( 'admin/seatsReport.js' ),
+			'admin/seatsReport.js',
 			Tickets_Main::VERSION
 		)
+			->add_to_group_path( 'tec-seating' )
+			->prefix_asset_directory( false )
```
Comment (lucatume) [inline src/Tickets/Seating/Orders/Controller.php:535]: Having to call a second method, `prefix_asset_directory`, on each registration looks tiring and error-prone.
My understanding is the group, `tec-seating`, and whether the `prefix_asset_directory` should be true or not are coupled, why not make the `prefix_asset_directory` a **group** property, not an asset property?

File: src/Tickets/Seating/Asset_Group_Path.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	/**
+	 * Registers the asset group path for seating
+	 *
+	 * @since 5.16.0
+	 *
+	 * @return void The asset group path is registered.
+	 */
+	protected function do_register(): void {
+		Config::add_group_path( 'tec-seating', Tickets_Plugin::instance()->plugin_path . 'build/', 'Seating/' );
```
Comment (lucatume) [inline src/Tickets/Seating/Asset_Group_Path.php:44]: See my comment about the `prefix_asset_path` method hiding what, I think, should be a group property, not an asset one.

**Review by lucatume — CHANGES_REQUESTED** [review]
Left some questions about how I think the API could be improved.
Requested some changes about test readability.

File: src/Tickets/Seating/Orders/Controller.php (hunk 2 above)
Comment (dpanta94) [reply src/Tickets/Seating/Orders/Controller.php:535]: Completely agree - Wanted to fix it when i discovered that was needed on Friday but there wasnt enough time. Will open a PR today amending that.

File: tests/slr_integration/Frontend/Timer_Test.php (hunk 1 above)
Comment (dpanta94) [reply tests/slr_integration/Frontend/Timer_Test.php:79]: I want to pass the slug and the path though. Doing that through a foreach seemed like the easier way.

File: src/Tickets/Seating/Asset_Group_Path.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+use Tribe__Tickets__Main as Tickets_Plugin;
+
+/**
+ * Class Asset_Group_Path.
+ *
+ * @since TBD
+ *
+ * @package TEC\Controller;
+ */
+class Asset_Group_Path extends Controller_Contract {
```
Comment (lucatume) [inline src/Tickets/Seating/Asset_Group_Path.php:23]: Why a dedicated class? It's a lot of boilerplate for one line of logic. Could this be moved to the `Assets` controller?

**Review by lucatume — CHANGES_REQUESTED** [review]
I think the `Asset_Group_Path` class could be eliminated.

File: src/Tickets/Seating/Asset_Group_Path.php (hunk 4 above)
Comment (dpanta94) [reply src/Tickets/Seating/Asset_Group_Path.php]: Yea makes sense. Added the single line i needed in the registration of the main seating controller.

File: jest.config.js [hunk 5]
```diff
@@ -11,5 +11,8 @@ module.exports = {
 		// Seating feature.
 		'^@tec/tickets/seating/tests/(.*)': '<rootDir>/tests/slr_jest/$1',
 		'^@tec/tickets/seating/(.*)': '<rootDir>/src/Tickets/Seating/app/$1',
+		// Waitlist feature.
```
Comment (lucatume) [inline jest.config.js:14]: Spill-over?

**Review by lucatume — APPROVED** [review]
Smaller left-over from other feature work

**Pull request merged**